opm/common/quad.hpp

```diff
@@ -0,0 +1,325 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+/*!
+ * \file
+ *
+ * \brief This file provides the infrastructure to use quad-precision
+ *        floating point values in the numerical models.
+ */
+#if !defined OPM_COMMON_QUAD_HPP && HAVE_QUAD
+#define OPM_COMMON_QUAD_HPP
+
+#ifdef DUNE_CLASSNAME_HH
+#error "Due to some trickery required for the linker, this file must be included _before_ Dune's classname.hh!"
+#endif
+
+#include <dune/common/classname.hh>
```

Comment:
> If you insist on keeping the classname specialization 

I'm not too dogmatic about this: I would not even mind about `Dune::className(quad)` producing some glibberish, but unfortunately it results in a linker error. So my position is that if we have to do something about __float128 anyway, I think we should make it as fool-proof as possible!?

> then I guess we should just let quad.hpp stay where it is currently (opm-material) until we find a better solution, or circumstances change. I would not mind that.

to me that's also okay. please note that with the `#if HAVE_DUNE_COMMON` solution, no dependency on dune-common is introduced because `quad.hh` is header-only...